docs/source/internal/generation_utils.rst

```diff
@@ -20,6 +20,90 @@ This page lists all the utility functions used by :meth:`~transformers.Pretraine
 
 Most of those are only useful if you are studying the code of the generate methods in the library.
 
+Generate Outputs
+-----------------------------------------------------------------------------------------------------------------------
+
+The output of :meth:`~transformers.PretrainedModel.generate` is an instance of a subclass of
+:class:`~transformers.file_utils.ModelOutput`. This output is a data structures containing all the information returned
+by :meth:`~transformers.PretrainedModel.generate`, but that can also be used as tuples or dictionaries.
+
+Let's see of this looks on an example:
+
+.. code-block::
+
+    from transformers import GPT2Tokenizer, GPT2LMHeadModel
+
+    tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
+    model = GPT2LMHeadModel.from_pretrained('gpt2')
+
+    inputs = tokenizer("Hello, my dog is cute and ", return_tensors="pt")
+    sequences = model.generate(**inputs, return_dict_in_generate=True, output_scores=True)
+
+The ``sequences`` object is a :class:`~transformers.generation_utils.GreedySearchDecoderOnlyOutput`, as we can see in
+the documentation of that class below, it means it has a ``sequences`` an optional ``scores``, an optional
+``hidden_states`` and an optional ``attentions`` attribute. Here we have the ``scores`` since we passed along
+``output_scores=True``, but we don't have ``hidden_states`` and ``attentions`` because we didn't pass
+``output_hidden_states=True`` or ``output_attentions=True``.
```

Comment:
I feel like this could be made easier to read by spreading it over a list:

```suggestion
The ``sequences`` object is a :class:`~transformers.generation_utils.GreedySearchDecoderOnlyOutput`, as we can see in
the documentation of that class below, it means it has the following attributes:

- a ``sequences``
- an optional ``scores``
- an optional ``hidden_states``
- an optional ``attentions``

Here we have the ``scores`` since we passed along ``output_scores=True``, but we don't have ``hidden_states`` and ``attentions`` because we didn't pass ``output_hidden_states=True`` or ``output_attentions=True``.
```